app/api/hippo/openai/[...path]/route.ts

```diff
@@ -0,0 +1,116 @@
+import { type OpenAIListModelResponse } from "@/app/client/platforms/openai";
+import { getServerSideConfig } from "@/app/config/server";
+import { ModelProvider, OpenaiPath } from "@/app/constant";
+import { prettyObject } from "@/app/utils/format";
+import { NextRequest, NextResponse } from "next/server";
+import { auth } from "../../../auth";
+import { requestOpenai } from "../../../common";
+import { genRagMsg } from "../../.common/rag";
+
+const ALLOWD_PATH = new Set(Object.values(OpenaiPath));
+
+function getModels(remoteModelRes: OpenAIListModelResponse) {
+  const config = getServerSideConfig();
+
+  if (config.disableGPT4) {
+    remoteModelRes.data = remoteModelRes.data.filter(
+      (m) => !m.id.startsWith("gpt-4"),
+    );
+  }
+
+  return remoteModelRes;
+}
+// this is hippo req
+async function convertNextChatRequest(req: NextRequest) {
+  let reqBody = await req.json();
+  let messages = reqBody["messages"];
+  let lastMessage = messages[messages.length - 1];
+  lastMessage.content = await genRagMsg(lastMessage, reqBody["userId"]);
+  delete reqBody["userId"];
```

Comment:
"userId" is only used as an index for data stored in the database, and openAI has been configured not to accept any other data fields (userId is something I added myself), otherwise it will fail. delete before sending to openAI.